File: test_gemini.py
import os
from dotenv import load_dotenv
from langchain_google_genai import ChatGoogleGenerativeAI

# Load environment variables
load_dotenv()

def test_gemini_api():
    """
    Test if the Gemini API is working and can chat with it.
    """
    # Get API key from environment or use a default one
    google_api_key = os.getenv("GOOGLE_API_KEY")
    
    if not google_api_key:
        print("Error: No Google API key found in environment variables.")
        print("Please set the GOOGLE_API_KEY environment variable.")
        return False
    
    try:
        # Initialize the Gemini model with the same settings as in the main app
        chat_model = ChatGoogleGenerativeAI(
            model="gemini-2.0-flash",  # Using the same model as in settings
            google_api_key=google_api_key,
            temperature=0.2,
            convert_system_message_to_human=True
            # Removed safety_settings as the format has changed in the latest version
        )
        
        # Test with a simple query
        response = chat_model.invoke("Hello! Can you tell me what you can do to help with financial advice?")
        
        print("\n=== Gemini API Test Results ===")
        print("✅ Connection successful!")
        print("\nResponse from Gemini:")
        print(response.content)
        print("\n===============================")
        
        return True
    
    except Exception as e:
        print("\n=== Gemini API Test Results ===")
        print("❌ Connection failed!")
        print(f"Error: {str(e)}")
        print("\n===============================")
        
        return False

if __name__ == "__main__":
    print("Testing Gemini API connection...")
    test_gemini_api()

File: backend/test_gemini.js
/**
 * Test file for Gemini AI integration
 * Run with: node test_gemini.js
 */

import { ChatGoogleGenerativeAI } from '@langchain/google-genai';
import dotenv from 'dotenv';

// Load environment variables
dotenv.config();

// Test query
const testQuery = "What are the best strategies for long-term investing?";

async function testGemini() {
  try {
    console.log("Testing Gemini AI integration...");
    
    const apiKey = process.env.GOOGLE_API_KEY;
    if (!apiKey) {
      console.error("ERROR: No Gemini API key found in environment variables!");
      console.error("Please set GOOGLE_API_KEY in your .env file");
      return;
    }
    
    console.log("API Key found, initializing Gemini model...");
    
    const model = new ChatGoogleGenerativeAI({
      model: process.env.GEMINI_MODEL || "gemini-2.0-flash",
      apiKey: apiKey,
      temperature: 0.2,
      convertSystemMessageToHuman: true
    });
    
    console.log("Sending test query to Gemini...");
    console.log(`Query: "${testQuery}"`);
    
    const response = await model.invoke(testQuery);
    
    console.log("\n----- GEMINI RESPONSE -----");
    console.log(response.content);
    console.log("--------------------------\n");
    
    console.log("Gemini test completed successfully!");
  } catch (error) {
    console.error("Gemini test failed with error:", error);
    console.error("Error message:", error.message);
    if (error.response) {
      console.error("Response data:", error.response.data);
    }
  }
}

// Run the test
testGemini();
